app/forms.py

```diff
@@ -15,3 +17,48 @@ def __init__(self, *args, **kwargs):
         super(ProfileUpdateForm, self).__init__(*args, **kwargs)
         self.fields["tags"].widget = CheckboxSelectMultiple()
         self.fields["tags"].queryset = Tag.objects.all()
+
+
+class SocialLinkForm(ModelForm):
+    class Meta:
+        model = SocialLinkAttachement
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        if "obj" in kwargs:
+            self.object = kwargs.pop("obj")
+
+        super(SocialLinkForm, self).__init__(*args, **kwargs)
+        self.fields["content_type"].widget = HiddenInput()
+        self.fields["object_id"].widget = HiddenInput()
+
+        if getattr(self, "object", None) is not None:
+            self.fields["content_type"].initial = ContentType.objects.get_for_model(
+                self.object.__class__
+            )
+
+    def clean_object_id(self):
+        cleaned_data = self.cleaned_data
+        if getattr(self, "object", None) is not None:
+            cleaned_data["object_id"] = self.object.pk
+        elif self.fields["object_id"].initial is not None:
+            cleaned_data["object_id"] = self.fields["object_id"].initial
+        return cleaned_data["object_id"]
```

Comment:
This overrides whatever value the user set for the (hidden) `object_id` field, in case one decided to manually edit its value. It's also used to set the aforementioned field's value if it was part of an object creation form. 
1. Should the value be simply overridden or should an error appear if the user attempts to manually edit the input value? 
2. Is there a better way to do the latter point?

Reply:
> This overrides whatever value the user set for the (hidden) object_id field, in case one decided to manually edit its value.

This can be achieved by making the field `disabled` (`field.disabled=True`) instead.